yt_dlp/extractor/youtube.py

```diff
@@ -2643,30 +2643,53 @@ def _extract_signature_timestamp(self, video_id, player_url, ytcfg=None, fatal=F
         return sts
 
     def _mark_watched(self, video_id, player_responses):
-        playback_url = get_first(
-            player_responses, ('playbackTracking', 'videostatsPlaybackUrl', 'baseUrl'),
-            expected_type=url_or_none)
-        if not playback_url:
+        mark_watched_urls = {
+            "playback": get_first(
+                player_responses, ('playbackTracking', 'videostatsPlaybackUrl', 'baseUrl'),
+                expected_type=url_or_none),
+            "watchtime": get_first(
+                player_responses, ('playbackTracking', 'videostatsWatchtimeUrl', 'baseUrl'),
+                expected_type=url_or_none)
+        }
```

Comment:
unwrapping it nearly doubles the amount of code because you need to extract the query params of each url, etc. i'll see what i can refactor into some helper functions to avoid that though

Reply:
Totally my bad, ignore this!